node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -254,34 +254,40 @@ case class DataMessageHandler(
         }
       case msg: BlockMessage =>
         val block = msg.block
-        logger.info(
-          s"Received block message with hash ${block.blockHeader.hash.flip.hex}")
 
-        val newApiF = {
-          chainApi
-            .getHeader(block.blockHeader.hashBE)
-            .flatMap { headerOpt =>
-              if (headerOpt.isEmpty) {
-                logger.debug("Processing block's header...")
-                for {
-                  processedApi <- chainApi.processHeader(block.blockHeader)
-                  _ <-
-                    appConfig.nodeCallbacks
-                      .executeOnBlockHeadersReceivedCallbacks(
-                        logger,
-                        Vector(block.blockHeader))
-                } yield processedApi
-              } else Future.successful(chainApi)
-            }
-        }
+        if (syncing) {
```

Comment:
This might need to be thought about a bit more, what happens in this case? 

1. Block headers are done syncing
2. Filter headers are done syncing
3. We are syncing filters, and one of those filters happens to match in the wallet, and we request the full block. Does this mean we ignore this block?